### PR TITLE
spacy language pipe documents processing

### PR DESCRIPTION
**Pull request by feedmari**
**Proposed changes**:
- Spacy offers the possibility to use Language.pipe method in order to process texts as a stream  and yield Doc objects in order. This is usually more efficient than processing texts one-by-one

Reference: https://spacy.io/api/language#pipe 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for submitting this PR we'll give it a review as soon as possible!

**Comment by feedmari** [comment]
> Thanks for submitting this PR we'll give it a review as soon as possible!

Ok, let me know! This fix make all the training process far away faster than before :)

File: rasa/nlu/utils/spacy_utils.py [hunk 1]
```diff
@@ -101,12 +101,23 @@ def doc_for_text(self, text: Text) -> "Doc":
...
+
+        return docs 
+
     def train(
         self, training_data: TrainingData, config: RasaNLUModelConfig, **kwargs: Any
     ) -> None:
 
-        for example in training_data.training_examples:
-            example.set("spacy_doc", self.doc_for_text(example.text))
+    	docs = self.docs_for_training_data(training_data)
```
Comment (tabergma) [inline rasa/nlu/utils/spacy_utils.py]:
```suggestion
        docs = self.docs_for_training_data(training_data)
```

Comment (tabergma) [reply rasa/nlu/utils/spacy_utils.py]: Incorrect formatting. That is also the reason why travis is failing.

File: rasa/nlu/utils/spacy_utils.py [hunk 2]
```diff
@@ -101,12 +101,23 @@ def doc_for_text(self, text: Text) -> "Doc":
         else:
             return self.nlp(text.lower())
 
+    def docs_for_training_data(self, training_data: TrainingData) -> List[Any]:
+
+        texts = [e.text if self.component_config.get("case_sensitive") else e.text.lower()
```
Comment (tabergma) [inline rasa/nlu/utils/spacy_utils.py]:
```suggestion
        texts = [
```

File: rasa/nlu/utils/spacy_utils.py [hunk 3]
```diff
@@ -101,12 +101,23 @@ def doc_for_text(self, text: Text) -> "Doc":
         else:
             return self.nlp(text.lower())
 
+    def docs_for_training_data(self, training_data: TrainingData) -> List[Any]:
+
+        texts = [e.text if self.component_config.get("case_sensitive") else e.text.lower() 
+                    for e in training_data.intent_examples]
```
Comment (tabergma) [inline rasa/nlu/utils/spacy_utils.py]:
```suggestion
            e.text if self.component_config.get("case_sensitive") else e.text.lower()
            for e in training_data.intent_examples
        ]
```

File: rasa/nlu/utils/spacy_utils.py [hunk 4]
```diff
@@ -101,12 +101,23 @@ def doc_for_text(self, text: Text) -> "Doc":
...
+
+        docs = [doc for doc in self.nlp.pipe(texts, batch_size=50)]
+
+        return docs
```
Comment (tabergma) [inline rasa/nlu/utils/spacy_utils.py]:
```suggestion
        return docs
```

**Comment by feedmari** [comment]
@tabergma Thanks for helping me with trevis :)

**Pull request merged**